### PR TITLE
Update IgnoreNullValues obsoletion message

### PR DESCRIPTION
**Pull request by layomia**
Should have been part of https://github.com/dotnet/runtime/pull/39388.

### CODE REVIEW

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs [hunk 1]
```diff
@@ -208,7 +208,7 @@ public int DefaultBufferSize
         /// Thrown if this property is set after serialization or deserialization has occurred.
         /// or <see cref="DefaultIgnoreCondition"/> has been set to a non-default value. These properties cannot be used together.
         /// </exception>
-        [Obsolete("To ignore null values when serializing, set DefaultIgnoreCondition to JsonIgnoreCondition.WhenWritingDefault.", error: false)]
+        [Obsolete("To ignore null values when serializing, set DefaultIgnoreCondition to JsonIgnoreCondition.WhenWritingNull.", error: false)]
```
[the remark this answers is not in the document]
Comment (layomia) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs:211]: Yes I believe it should. @safern, I had to manually add the attributes in https://github.com/dotnet/runtime/pull/41141/commits/3989b19131120e98d5065afdf6f81fd4062ce17d; using `/t:GenerateReferenceSource` did not update the property with the attributes. Is this expected?

[the remark this answers is not in the document]
Comment (safern) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs:211]: `ObsoleteAttribute` is part of: https://github.com/dotnet/runtime/blob/master/eng/DefaultGenApiDocIds.txt#L57 which causes the tools to ignore these attributes. We haven't enforced this attribute on both the ref and the src, we just care about it on the ref. However, I think we should start enforcing that as that could potentially cause miss matches or us forgetting to include them on the ref.

**Review by safern — APPROVED** [review]

**Comment by jeffhandley** [comment]
The obsoletion breaking change will be covered by dotnet/docs#23856.

**Pull request merged**